proxy/http2/Http2DependencyTree.h

```diff
@@ -155,23 +155,43 @@ Http2DependencyTree<T>::_find(Node *node, uint32_t id, uint32_t depth)
 }
 
 template <typename T>
-typename Http2DependencyTree<T>::Node *
-Http2DependencyTree<T>::find(uint32_t id)
+Node *
+Tree<T>::find_shadow(uint32_t id)
 {
   return _find(_root, id);
 }
 
 template <typename T>
-typename Http2DependencyTree<T>::Node *
-Http2DependencyTree<T>::add(uint32_t parent_id, uint32_t id, uint32_t weight, bool exclusive, T t)
+Node *
+Tree<T>::find(uint32_t id)
+{
+  Node *n = _find(_root, id);
+  return n == nullptr ? nullptr : (n->is_shadow() ? nullptr : n);
+}
+
+template <typename T>
+Node *
+Tree<T>::add(uint32_t parent_id, uint32_t id, uint32_t weight, bool exclusive, T t)
 {
   Node *parent = find(parent_id);
-  if (parent == NULL) {
-    parent = _root;
+  if (parent == nullptr) {
+    if (exclusive) {
+      parent = add(0, parent_id, weight, false, nullptr);
+    } else {
+      parent = _root;
```

Comment:
Sorry, one more question. Why don't we need to create shadow node if exclusive flag is false?
In the original issue, the example is using `exclusive = true`, but it looks like RFC 7540 doesn't restrict the condition of exclusive flag.

> A dependency on a stream that is not currently in the tree -- such as a
stream in the "idle" state -- results in that stream being given a
default priority (Section 5.3.5).